docs/conceptual/iommu.rst

```diff
@@ -0,0 +1,98 @@
+.. meta::
+   :description: Input-Output Memory Management Unit (IOMMU)
+   :keywords: IOMMU, DMA, PCIe, XGMI, AMD, ROCm
+
+****************************************************************
+Input-Output Memory Management Unit (IOMMU)
+****************************************************************
+
+Overview
+================================================================
+
+The IOMMU (IO Memory Management Unit) is a device which provides memory remapping services to 
+IO devices on a platform. Similar to the MMU on a CPU, the IOMMU is able to provide a private 
+virtual address space for each device or group of devices on a platform. While the IOMMU is 
+most commonly used for virtualization provide a private IO space for devices used in a guest 
+virtual machine, it can also be used to isolate devices on bare metal to prevent accidental
+or nefarious DMA accesses to things that they should should not have access to. The IOMMU provides
+virtualized access to the entire platform address space, not just system memory. It can also
+provide virtualized access to the MMIO portions of the platform address space. This is used for
+things like peer to peer DMA. In addition to DMA remapping, the IOMMU also support interrupt remapping.
+In addition to virtualization use cases, this can be advantageous for devices which support multiple
+interrupts on bare metal and it may be required for interrupt delivery on platforms with lots of cores.
+
+
+Peer to Peer DMA
+================================================================
+
+Peer to Peer DMA over PCIe
+----------------------------------------------------------------
+
+Peer to Peer DMA is not officially part of the PCI/PCIe spec. As such, the behavior for peer to peer DMA
+varies from platform to platform. AMD platforms prior to zen only supported peer to peer DMA for writes.
+On Zen and newer CPUs, peer to peer DMA is fully supported. On Intel, peer to peer DMA is supported with
```

Comment:
Is the behaviour on Intel impotrtant here?

Reply:
Maybe say "AMD Zen" rather than just "Zen".  As for Intel, I thought it was useful to call out the limitations since customers may be using Intel platforms.